test/index.js

```diff
@@ -93,6 +93,29 @@ describe( 'I18n', function() {
 					count: 2
 				} ) );
 			} );
+
+			describe( 'adding new translations', function() {
+				it( 'should find a new translation after it has been added', function() {
+					i18n.addTranslations( {
+						'test3': [
+							null,
+							'translation3'
+						]
+					} );
+
+					assert.equal( 'translation3', translate( 'test3' ) );
+				} );
+				it( 'should return the new translation if it has been overwritten', function() {
+					i18n.addTranslations( {
+						'test1': [
+							null,
+							'not-translation1'
+						]
+					} );
+
+					assert.equal( 'not-translation1', translate( 'test1' ) );
+				} );
+			} );
 		} );
```

Comment:
"adding new translations" is underneath "plural translation" here, should probably be under `I18n`.

It'd also be nice to have a test that we don't unintentionally trash the other settings under `''`.